### DISCUSSION TITLE
2.04 Card Trick Task 2

### DISCUSSION THREAD

**Discussion by jfhr990120** [rramosp/ai4eng.v1#312] [02 - PYTHON]
Hola buenas.
Lo que pasa es que estaba haciendo la Task 2 del Card Trick y obtengo una respuesta trivial de la plataforma

Esta es mi función

```
def collect_heaps(h, n):
    r= h.copy() #Las copio para tener la estructura
    h_no_modi =h.copy() #la copio para después utilizar las filas enteras en el arreglo
    h = np.array(h) #La vuelvo array para manipularla y buscar con NumPy
    buscar = np.where(h == n)[0] #La fila que va en el centro
    heaps_mix = np.arange(0,h.shape[0]) #Hago un array para hacer el shuffling
    np.random.shuffle(heaps_mix) #Acá ordeno de manera aleatoria
    while heaps_mix[h.shape[0]//2] != buscar: #shuffleo las posiciones
      np.random.shuffle(heaps_mix)
    
    r = [h_no_modi[i] for i in heaps_mix] #Ordeno según el array de posiciones del original

    return r
```

Y en las pruebas manuales funciona bien, pero no en las herramienta de calificación, pues obtengo este error:

![image](https://github.com/rramosp/ai4eng.v1/assets/162035477/6ceedc4a-99f7-4dc5-bff5-fb17de80f701)

Se me hace muy extraño porque mi respuesta sí imprime el heap correcto en la mitad pero la plataforma no la lee

@Felipe-RA ayuda por favor

**Comment by jfhr990120** [comment]
Solucionado

**Reply by Felipe-RA** [reply]
Hola @jfhr990120 

Me alegra que lo hayas podido solucionar.

Si no te molesta,   ¿podrías comentarnos qué encontraste que le faltaba a tu respuesta? 

Gracias por tu tiempo!

**Reply by jfhr990120** [reply]
Hola, muchas gracias por la respuesta
En realidad estaba imprimiendo la respuesta en un array distinto al formato que pedía la plataforma
Inicialmente mi código armaba un array de la forma [[heap_1], [heap_medio], [heap2]] (es un ejemplo para el caso de 3) porque yo creí que la plataforma iba a mirar un arreglo de posiciones impares e iba a buscar la de la mitad. 
Pero no, lo que hace el código de la plataforma (según entiendo) es tomar los heaps de la mitad de una única lista. Así, para el caso de 3 heaps cada uno de 7 cartas, lo que hace la plataforma es buscar en una lista las posiciones 8 a 14 (índices 7 a 13) y compararlas con el arreglo interno que debería de dar.

Entonces cambié mi código para que en vez de un array de arrays (como está descrito arriba) fuese un único array con la secuencia completa de cartas
